src/pytest_inmanta_lsm/managed_service_instance.py

```diff
@@ -349,22 +350,19 @@ def check_start_state(current_state: State) -> bool:
                 return False
             return current_state.version == start_version
 
-        def get_bad_state_error(current_state) -> Any:
-            validation_failure_msg = self.remote_orchestrator.get_validation_failure_message(
-                service_entity_name=self.service_entity_name,
-                service_instance_id=self.instance_id,
+        def get_bad_state_error(current_state: State) -> FullDiagnosis:
+            result = self.remote_orchestrator.client.lsm_services_diagnose(
+                tid=self.remote_orchestrator.environment,
+                service_entity=self.service_entity_name,
+                service_id=self.instance_id,
+                version=current_state.version,
             )
-            if validation_failure_msg:
-                return validation_failure_msg
-
-            LOGGER.info("No validation failure message, getting failed resource logs")
-
-            # No validation failure message, so getting failed resource logs
-            failed_resource_logs = FailedResourcesLogs(
-                self.remote_orchestrator.client,
-                self.remote_orchestrator.environment,
+            assert result.code == 200, (
+                f"Wrong response code while trying to get the service diagnostic, got {result.code} (expected 200):\n"
+                f"{json.dumps(result.result or {}, indent=4)}"
```

Comment:
The ticket mentions that a fallback should be done to the old code when the diangose API endpoint is not available. That is not implemented here, but I guess it's also not requires, because all ISO versions have the diagnose endpoint now?

Reply:
Iso3 might not have it but we don't run any tests against any iso3 orchestrator anyway